paddle/gserver/layers/PixelSoftmaxLayer.cpp

```diff
@@ -0,0 +1,89 @@
+/* Copyright (c) 2016 PaddlePaddle Authors. All Rights Reserve.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License. */
+
+#include "PixelSoftmaxLayer.h"
+#include "paddle/utils/Stat.h"
+
+namespace paddle {
+
+REGISTER_LAYER(pixel_softmax, PixelSoftmaxLayer);
+
+bool PixelSoftmaxLayer::init(const LayerMap& layerMap,
+                             const ParameterMap& parameterMap) {
+  /* Initialize the basic parent class */
+  Layer::init(layerMap, parameterMap);
+  auto& img_conf = config_.inputs(0).image_conf();
+  inH_ =
+      img_conf.has_img_size_y() ? img_conf.img_size_y() : img_conf.img_size();
+  inW_ = img_conf.img_size();
+  inC_ = img_conf.channels();
+  createFunction(forward_, "NCHW2NHWC", FuncConfig());
+  createFunction(backward_, "NHWC2NCHW", FuncConfig());
```

Comment:
I can't rename forward_ and backward_ directly .  Because they are defined in Layer.h. So i added `std::vector<std::shared_ptr<FunctionBase>> nchw2nhwc_;` and `std::vector<std::shared_ptr<FunctionBase>> nhwc2nchw_;` into PixelSoftmaxLayer.h to fix this problem. Thx for your suggestion.